### PR TITLE
ext: mcux: Add import script and import latest version

### LINKED ISSUE #5833: Script to import mcux sdk
New versions of the mcux sdk need to be imported whenever a new nxp soc is added to zephyr, or when nxp releases updated mcux drivers for an existing soc. Create a script to automate importing the mcux sdk into zephyr.

### PR DESCRIPTION
**Pull request by MaureenHelm**
Adds a new python script to automate importing the MCUXpresso SDK into Zephyr and uses the script to import the latest version. Parts of the SDK are re-licensed with a BSD-3-Clause-Clear license, which is a new license to Zephyr.

Updates [README](https://github.com/codeauroraforum/zephyr/blob/mcux-update/ext/hal/nxp/mcux/README) to follow template and reflect new licensing.

Fixes #5833

@mike-scott @vakulgarg The new uart driver supports hw flow control

### CODE REVIEW

**Review by galak — COMMENTED** [review]
On the import script, would be good maybe in the commit message to show an example invocation of the script.

File: ext/hal/nxp/mcux/drivers/imx/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2018, NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+zephyr_include_directories(.)
+
+zephyr_sources_ifdef(CONFIG_GPIO_MCUX_IGPIO   fsl_gpio.c)
```
Comment (galak) [inline ext/hal/nxp/mcux/drivers/imx/CMakeLists.txt]: Will drivers/imx be able to handle both imx-rt and imx6/7/8?

**Comment by mike-scott** [comment]
Thank you @MaureenHelm !   When I'm back in the office tomorrow, I'll rebase my LTE modem branch and test this update

**Comment by mike-scott** [comment]
@MaureenHelm: are you also setting up a PR for drivers/serial/uart_mcux.c to set enableRxRTS / enableTxCTS based on Kconfig options?

**Comment by jukkar** [comment]
Hi @MaureenHelm, there are couple of nxp hal fixes in gPTP PR #6570.
The patches are called:
* ext: hal: nxp: mcux: Fix ethernet timestamping driver
* ext: hal: nxp: mcux: Enable enhanced buffer desc mode if needed
* ext: hal: nxp: mcux: Fix PTP event packet type check

How could we get those fixes merged in upstream nxp hal?

**Comment by MaureenHelm** [comment]
> Hi @MaureenHelm, there are couple of nxp hal fixes in gPTP PR #6570.
> The patches are called:
> 
> ext: hal: nxp: mcux: Fix ethernet timestamping driver
> ext: hal: nxp: mcux: Enable enhanced buffer desc mode if needed
> ext: hal: nxp: mcux: Fix PTP event packet type check
> How could we get those fixes merged in upstream nxp hal?

Thanks @jukkar, I'll file an internal ticket. Only the first and third commits need to go upstream, the second modifies a file unique to Zephyr.

**Comment by mike-scott** [comment]
@MaureenHelm I tested this PR along with my current modem driver and everything seemed to work just fine.  As a part of that testing, I rebased a patch to add the Kconfig settings for HW Flow Control support here: https://github.com/mike-scott/zephyr/commit/a296fcc089a42653360e04438b1d03f3295b749a    Let me know if you want me to put that up as a PR linked to this one.  Or feel free to add it here if you want.

**Comment by galak** [comment]
@MaureenHelm any thoughts on my comments?  I asked, because I can see us wanting to merge the re-org change and related commits while we wait on any discussion about the updated import.  (Not sure how the re-org will impact PR #6367

File: ext/hal/nxp/mcux/drivers/imx/CMakeLists.txt (hunk 1 above)
Comment (MaureenHelm) [reply ext/hal/nxp/mcux/drivers/imx/CMakeLists.txt]: That was the idea, but I found an exception today. This works for rt, imx6, imx7d, and imx8; but not imx7ulp.

**Review by carlescufi — COMMENTED** [review]
In general all looks good, but maybe the import script should not be in the root of `scripts/`, but rather keep it somewhere in `scripts/support`?

**Comment by nashif** [comment]
> In general all looks good, but maybe the import script should not be in the root of scripts/, but rather keep it somewhere in scripts/support?

IMO this should be in ext/hal/nxp/mcux/scripts, all in one place and when we move HALs out of the tree at some point, the script will be part of the deal.

**Comment by MaureenHelm** [comment]
- Rebased to latest master
- Moved script to ext/hal/nxp/mcux/scripts
- Added example usage to commit message

**Comment by mike-scott** [comment]
Is this still stuck in license review hell?

**Comment by MaureenHelm** [comment]
> Is this still stuck in license review hell?

Yes. The governing board is still discussing it.

**Comment by MaureenHelm** [comment]
Rebased to latest master

**Comment by galak** [comment]
Do we want to pull the script out into its own PR, and close this for now?

**Comment by MaureenHelm** [comment]
> Do we want to pull the script out into its own PR, and close this for now?

Sure

**Pull request closed without merging**